designs/2018-programmatic-style-edits/README.md

```diff
@@ -0,0 +1,362 @@
+- Start Date: 2018-11-20
+- RFC PR: (leave this empty, to be filled in later)
+- Authors: Nicholas C. Zakas (@nzakas)
+
+# Programmatic Style Editors
+
+## Summary
+
+A way to extend the capabilities of ESLint's `--fix` command to enable applying  programmatic style edits instead of (or in addition to) using stylistic rules. Developers can create automatically applied style guides by writing JavaScript code rather than configuring individual rules.
+
+## Motivation
+
+The way ESLint has traditionally helped developers enforce style guides is through the implementation of individual rules. Each rule then needs to be manually configured by someone and the ESLint team ends up maintaining the rule. Because people enjoy using shareable configs, developers frequently ask for new style rules and new options to existing style rules be implemented in the core. This has lead to an explosion of stylistic rules that are getting more difficult to maintain and get to work seamlessly with one another (i.e., ensuring rules don't conflict with one another or overlap).
+
+Tools like [Prettier](https://prettier.io/) have taken a different approach, where there is a small set of configuration options available. Instead of warning on each small style disparity, it simply formats everything according to the configuration options. This opinionated approach is simpler to maintain but has the downside of being too strict for some developers who prefer more fine-grained control.
+
+Prettier has also shown that there is a desire to not fill up end-user screens with multiple error messages advising developers to add a space here or remove a space there; they'd much rather just have one command to fix the style of their source code.
+
+Additionally, a growing number of developers use ESLint together with Prettier to get both the error-checking capability of ESLint with the source code formatting of Prettier. While there are some approaches to making these two tools work together, any quick search of Twitter will find new developers tend to struggle to get these two tools to work together.
```

Comment:
Could you elaborate on what these usability issues are? The most frequent issue I've seen when people are using `eslint-plugin-prettier` is that they don't realize they need to disable other stylistic rules. But this seems like it would still be a problem if we introduced style editors.